### PR TITLE
Support for transparent pixels

### PR DESCRIPTION
**Pull request by Znunu**
### Description
Transparent pixels in the image will not be drawn. Semi transparent pixels will be rounded to either fully transparent or opaque, because I wasn't exactly sure what to do with semi transparent pixels. I could technically draw a semi transparent pixel by looking at the previous pixel, but that would make draw_and_fix go forever since there isn't really a clear stop condition.

### Implementation
The grid in AutoDrawer used be a 2D array of Pixels. I modified it to be an 2D array of tuples, each tuple containing a pixel and a boolean value. The boolean determines if the pixel is transparent or not i.e. if it's going to be drawn. 

### Test
I tested it with a few .pngs at scale=1 and it seems to work fine

### CODE REVIEW

File: dpypx/autodraw.py [hunk 1]
```diff
@@ -21,20 +21,16 @@ class AutoDrawer:
     @classmethod
     def load_image(
             cls, client: Client, xy: tuple[int, int],
-            image: Image.Image, scale: float = 1) -> AutoDrawer:
-        """Draw from the pixels of an image."""
```
Comment (Artemis21) [inline dpypx/autodraw.py:25]: Why remove the docstring?

File: dpypx/autodraw.py [hunk 2]
```diff
@@ -21,20 +21,16 @@ class AutoDrawer:
...
-        if image.mode == 'RGBA':
-            new_image = Image.new('RGB', image.size)
-            new_image.paste(image, mask=image)
-            image = new_image
+            image: Image.Image, scale: float = 1, strict: bool = False) -> AutoDrawer:
```
Comment (Artemis21) [inline dpypx/autodraw.py]: You added the `strict` param but don't seem to have used it...

File: dpypx/autodraw.py [hunk 3]
```diff
@@ -99,7 +95,11 @@ async def check_pixel(self, x: int, y: int) -> bool:
 
         Returns True if the pixel was not already drawn.
         """
-        colour = self.grid[y - self.y0][x - self.x0]
+        pixel = self.grid[y - self.y0][x - self.x0]
+        colour = pixel[0]
+        if not pixel[1]:
```
Comment (Artemis21) [inline dpypx/autodraw.py]:
```suggestion
        colour, opaque = self.grid[y - self.y0][x - self.x0]
        if not opaque:
```
Using iterable unpacking for shorter code and better readability.

File: dpypx/autodraw.py [hunk 4]
```diff
@@ -21,20 +21,16 @@ class AutoDrawer:
...
+        image = image.convert(mode="RGBA")
         width = round(image.width * scale)
         height = round(image.height * scale)
         # Since the image is only loaded once we can afford to use a high
         # quality filter.
         resized = image.resize((width, height), Image.LANCZOS)
         data = list(resized.getdata())
         grid = [
-            [Pixel(*pixel) for pixel in data[start:start + width]]
+            [(Pixel(*pixel[:3]), bool(round(pixel[3] / 255))) for pixel in data[start:start + width]]
```
Comment (Artemis21) [inline dpypx/autodraw.py]:
```suggestion
            [(Pixel(*pixel[:3]), pixel[3] > 127) for pixel in data[start:start + width]]
```
Equivalent code, less computation and simpler.

**Review by Artemis21 — CHANGES_REQUESTED** [review]
Minor issues, looks good though.

File: dpypx/autodraw.py (hunk 4 above)
Comment (Artemis21) [reply dpypx/autodraw.py]: Even so actually this line is longer than ideal. Perhaps
```suggestion
            [
                (Pixel(*pixel[:3]), pixel[3] > 127)
                for pixel in data[start:start + width]
            ]
```

**Comment by Znunu** [comment]
Thanks. Your suggestions were nice. Think I got them all.

File: dpypx/autodraw.py [hunk 5]
```diff
@@ -58,7 +55,7 @@ def load(cls, client: Client, data: str) -> AutoDrawer:
         for _ in range(height):
             row = []
             for _ in range(width):
-                row.append(Pixel.from_hex(lines.pop(0)))
+                row.append((Pixel.from_hex(lines.pop(0))), True)
```
Comment (Artemis21) [inline dpypx/autodraw.py]:
```suggestion
                row.append((Pixel.from_hex(lines.pop(0)), True))
```
I think the brackets are slightly misplaced here.

File: dpypx/autodraw.py [hunk 6]
```diff
@@ -23,18 +23,15 @@ def load_image(
...
+        image = image.convert(mode="RGBA")
         width = round(image.width * scale)
         height = round(image.height * scale)
         # Since the image is only loaded once we can afford to use a high
         # quality filter.
         resized = image.resize((width, height), Image.LANCZOS)
         data = list(resized.getdata())
         grid = [
-            [Pixel(*pixel) for pixel in data[start:start + width]]
+            [(Pixel(*pixel[:3]), pixel[3] > 127) for pixel in data[start:start + width]]
```
Comment (Artemis21) [inline dpypx/autodraw.py]: This line is 88 lines long but the repo tries to keep lines under 80 (I mentioned this when making the suggestion)

File: dpypx/autodraw.py [hunk 7]
```diff
@@ -99,7 +96,10 @@ async def check_pixel(self, x: int, y: int) -> bool:
 
         Returns True if the pixel was not already drawn.
         """
-        colour = self.grid[y - self.y0][x - self.x0]
+        colour, opaque = self.grid[y - self.y0][x - self.x0]
+	    if not opaque:
```
Comment (Artemis21) [inline dpypx/autodraw.py]:
```suggestion
        colour, opaque = self.grid[y - self.y0][x - self.x0]
	  if not opaque:
```
Indentation

**Review by Artemis21 — CHANGES_REQUESTED** [review]
Just one more thing

File: dpypx/autodraw.py (hunk 6 above)
Comment (Artemis21) [reply dpypx/autodraw.py]:
```suggestion
            [
                (Pixel(*pixel[:3]), pixel[3] > 127)
                for pixel in data[start:start + width]
            ]
```
Line length

**Review by Artemis21 — CHANGES_REQUESTED** [review]

**Pull request merged**